web/shared/src/typeahead.ts

```diff
@@ -44,6 +46,10 @@ type UnicodeEmoji = {
     emoji_code: string;
     reaction_type: "unicode_emoji";
     is_realm_emoji: false;
+    emoji_url?: string;
```

Comment:
It doesn't seem like `emoji_url` was used in this file, but it will be used by future code that uses `EmojiSuggestion`

Reply:
I feel like this is unlikely to be properly an optional field; either we are setting this value for every `UnicodeEmoji` this file interacts with, or none of them.

Reply:
The logic for making this optional came from `update_emoji_data` -- the type `EmojiDict` has an optional `url` field, and `emoji_collection` which is type `Emoji` is pushed an object without the `emoji_url` when `emoji_dict.is_realm_emoji` is `false.

I added `console.log(emoji_dict.is_realm_emoji, emoji_dict.url, emoji_dict.name)` to the for loop of this function and got `false undefined` for all the non-realm emoji, and this for the two realm emojis in the dev environment:

```
true /user_avatars/2/emoji/images/1.png green_tick
true /static/generated/emoji/images/emoji/unicode/zulip.png zulip
```

This confirms my understanding that what I put is the correct type given the way the code is currently set up.

Reply:
OK there's probably something to clean up there but I'm not sure what.